buffer.go

```diff
@@ -211,28 +200,12 @@ func (b *Buffer) dataCopy() ([]byte, error) {
 		return nil, nil
 	}
 
-	if b.data == nil {
-		// This is a TileDB-managed buffer. We need to copy its data into Go memory.
-		// We assume that once a buffer is set to point to user-provided memory,
-		// TileDB never updates the buffer to point to its own memory (i.e., the
-		// only time when there will be a buffer pointing to TileDB-owned memory is
-		// when TileDB allocates a fresh buffer, e.g. as an out parameter from a
-		// serialization function).
-
-		// Since this buffer is TileDB-managed, make sure it's not GC'd before we're
-		// done with its memory.
-		defer runtime.KeepAlive(b)
-
-		if csize > math.MaxInt32 {
-			return nil, fmt.Errorf("TileDB's buffer (%d) larger than maximum allowed CGo buffer (%d)", csize, math.MaxInt32)
-		}
-		return C.GoBytes(cbuffer, C.int(csize)), nil
+	if csize > math.MaxInt32 {
+		return nil, fmt.Errorf("TileDB's buffer (%d) larger than maximum allowed CGo buffer (%d)", csize, math.MaxInt32)
 	}
+	cpy := C.GoBytes(cbuffer, C.int(csize))
```

Comment:
We can't know whether the memory is owned by Go or C, so we always make a copy. This API is not recommended either way.